### PR TITLE
Drop ReconVarIC

### PR DESCRIPTION
**Pull request by permcody**
Get rid of it!

refs #9060 

@dschwen - This branch just builds off #9060 (one more commit - but it's a good one!)

### CODE REVIEW

**Comment by permcody** [comment]
@dschwen - ping

Also - note the line reduction and I'm not done yet!

File: modules/phase_field/examples/ebsd_reconstruction/IN100-111grn.i [hunk 1]
```diff
@@ -10,9 +10,29 @@
...
+    flood_entity_type = ELEMENTAL
+    compute_halo_maps = true # For displaying HALO fields
+    polycrystal_ic_uo = ebsd
+  [../]
+[]
+
+[ICs]
+  [./PolycrystalICs]
+    [./PolycrystalColoringIC]
+      polycrystal_ic_uo = ebsd
```
Comment (aeslaughter) [inline modules/phase_field/examples/ebsd_reconstruction/IN100-111grn.i]: Cody what is this about?

**Review by aeslaughter — DISMISSED** [review]
This looks great.

**Review by dschwen — APPROVED** [review]

File: framework/src/userobject/UserObject.C [hunk 2]
```diff
@@ -36,12 +36,18 @@ validParams<UserObject>()
                         "in the case this is true but no "
                         "displacements are provided in the Mesh block "
                         "the undisplaced mesh will still be used.");
-  params.addParamNamesToGroup("use_displaced_mesh", "Advanced");
+  params.addParam<bool>("allow_duplicate_execution_on_initial",
+                        false,
+                        "In the case where this UserObject is depended upon by an initial "
+                        "condition, allow it to be executed twice during the initial setup (once "
+                        "before the IC and again after mesh adaptivity (if applicable).");
```
Comment (permcody) [inline framework/src/userobject/UserObject.C]: It doesn't need to re-execute on each initial adaptivity step. The adaptivity steps will project the initial condition each time to the new mesh. When that's all done, we just need to rerun it once on the new mesh and deal with the changes. That's what's being done now. Most of the grain growth examples have 2 adaptivity steps, and I have examined the results and everything looks exactly correct.

**Pull request merged**